lib/mistral_ai_summary_example/mistralai_summary_page.dart

```diff
@@ -51,84 +49,159 @@ class _MistralAISummaryPageState extends State<MistralAISummaryPage> {
       setState(() {
         summaryResult = 'Error: $e';
       });
+    } finally {
+      setState(() {
+        summaryInProgress = false;
+      });
     }
   }
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text('MistralAI Summary'),
-        actions: [
-          IconButton(
-            onPressed: () async {
-              final newSettings = await Navigator.push(
-                context,
-                MaterialPageRoute<SummarySettings>(
-                  builder: (BuildContext context) => SettingsWidget(
-                    initialSettings: summarySettings,
-                  ),
-                ),
-              );
-              setState(() {
-                summarySettings = newSettings ?? summarySettings;
-              });
-            },
-            icon: const Icon(Icons.settings),
+    return ChangeNotifierProvider(
+      create: (_) => SummarySettingsModel(),
+      builder: (context, _) {
+        return Scaffold(
+          appBar: AppBar(
+            title: const Text('MistralAI Summary'),
```

Comment:
Just to match what we have on home screen :)
```suggestion
            title: const Text('MistralAI Text Summary'),
```